go-controller/pkg/ovn/controller/apbroute/repair.go

```diff
@@ -362,3 +378,68 @@ func (c *ExternalGatewayMasterController) buildOVNECMPCache() map[string][]*ovnR
 	}
 	return ovnRouteCache
 }
+
+// returns a hybrid cache of map[podIPs]nodeName
+func (c *ExternalGatewayMasterController) buildOVNHybridCache() map[string]string {
+	ovnHybridCache := make(map[string]string)
+	p := func(item *nbdb.LogicalRouterPolicy) bool {
+		return item.Priority == types.HybridOverlayReroutePriority
+	}
+
+	logicalRouterPolicies, err := c.nbClient.findLogicalRouterPoliciesWithPredicate(p)
+	if err != nil {
+		klog.Errorf("CleanHybridPRoutes: failed to list hybrid routes: %v", err)
+		return nil
+	}
+
+	for _, lrp := range logicalRouterPolicies {
+		// get node name
+		var nodeName string
+		for _, nexthop := range lrp.Nexthops {
+			p := func(item *nbdb.LogicalRouterPort) bool {
```

Comment:
Why don't you build a set of nexthops and run the logical port predicate once to output a set of node names

Reply:
sure I can do that